[2025-07-11T14:25:35.090+0000] {taskinstance.py:1125} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: final_project.Stage_songs manual__2025-07-11T14:25:32.608336+00:00 [queued]>
[2025-07-11T14:25:35.101+0000] {taskinstance.py:1125} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: final_project.Stage_songs manual__2025-07-11T14:25:32.608336+00:00 [queued]>
[2025-07-11T14:25:35.103+0000] {taskinstance.py:1331} INFO - Starting attempt 1 of 1
[2025-07-11T14:25:35.123+0000] {taskinstance.py:1350} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2025-07-11 14:25:32.608336+00:00
[2025-07-11T14:25:35.130+0000] {standard_task_runner.py:57} INFO - Started process 383 to run task
[2025-07-11T14:25:35.134+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'final_project', 'Stage_songs', 'manual__2025-07-11T14:25:32.608336+00:00', '--job-id', '14', '--raw', '--subdir', 'DAGS_FOLDER/final_project.py', '--cfg-path', '/tmp/tmpmeiwc4b5']
[2025-07-11T14:25:35.139+0000] {standard_task_runner.py:85} INFO - Job 14: Subtask Stage_songs
[2025-07-11T14:25:35.246+0000] {task_command.py:410} INFO - Running <TaskInstance: final_project.Stage_songs manual__2025-07-11T14:25:32.608336+00:00 [running]> on host 49f24e24a176
[2025-07-11T14:25:35.432+0000] {taskinstance.py:1570} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='udacity' AIRFLOW_CTX_DAG_ID='final_project' AIRFLOW_CTX_TASK_ID='Stage_songs' AIRFLOW_CTX_EXECUTION_DATE='2025-07-11T14:25:32.608336+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-11T14:25:32.608336+00:00'
[2025-07-11T14:25:35.454+0000] {base.py:73} INFO - Using connection ID 'aws_credentials' for task execution.
[2025-07-11T14:25:35.454+0000] {connection_wrapper.py:337} INFO - AWS Connection (conn_id='aws_credentials', conn_type='aws') credentials retrieved from login and password.
[2025-07-11T14:25:35.465+0000] {stage_redshift.py:53} INFO - Clearing data from destination Redshift table
[2025-07-11T14:25:35.476+0000] {base.py:73} INFO - Using connection ID 'redshift' for task execution.
[2025-07-11T14:25:37.885+0000] {sql.py:375} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2025-07-11T14:25:38.245+0000] {taskinstance.py:1847} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 54, in execute
    redshift.run("DELETE FROM {}".format(self.table))
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 349, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 380, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.UndefinedTable: relation "staging_songs" does not exist

[2025-07-11T14:25:38.265+0000] {taskinstance.py:1373} INFO - Marking task as FAILED. dag_id=final_project, task_id=Stage_songs, execution_date=20250711T142532, start_date=20250711T142535, end_date=20250711T142538
[2025-07-11T14:25:38.281+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 14 for task Stage_songs (relation "staging_songs" does not exist
; 383)
[2025-07-11T14:25:38.305+0000] {local_task_job_runner.py:232} INFO - Task exited with return code 1
[2025-07-11T14:25:38.387+0000] {taskinstance.py:2674} INFO - 0 downstream tasks scheduled from follow-on schedule check
